packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -345,7 +361,7 @@ export default function makeKernelKeeper(kvStore, streamStore, kernelSlog) {
         p.subscribers = commaSplit(kvStore.get(`${kernelSlot}.subscribers`));
         p.queue = Array.from(
           kvStore.getPrefixedValues(`${kernelSlot}.queue.`),
-        ).map(JSON.parse);
+        ).map(s => JSON.parse(s));
```

Comment:
I don't understand the motivation for this change.

Reply:
`JSON.parse` takes more than 1 arg (3, I think) and map provides more than 1 arg (double-checking... yes, 3). These extra args don't match up, type wise.

I could `@ts-ignore` it instead, but I gather that's not wise:

 - [Don't use functions as callbacks unless they're designed for it \- JakeArchibald\.com](https://jakearchibald.com/2021/function-callback-risks/)

I don't have a strong preference. I'm interested in your thoughts.

Reply:
@warner writes:
> given that both inputs and outputs have surprising/easily-forgettable arguments, yes, please retain the `s => JSON.parse(s)`

Reply:
Very much strongly in favor not using functions as callbacks when they're not meant to be. At least here tsc complains, but imagine if the type of the `JSON.parse` arguments had matched `map`'s callback.

Reply:
I hadn't realized that `map` feeds in more than one arg.  Given that it does, Dan's transform on this line is exactly right and my non-understanding has been resolved.